tt_metal/host_api.hpp

```diff
@@ -189,7 +189,7 @@ void UpdateDynamicCircularBufferAddress(Program &program, CBHandle cb_handle, co
  * | core_spec     | Range of the Tensix co-ordinates using the semaphore | const std::variant<CoreRange,CoreRangeSet> &              |              | Yes      |
  * | initial_value | Initial value of the semaphore                       | uint32_t                                                  |              | Yes      |
  */
-uint32_t CreateSemaphore(Program &program, const std::variant<CoreRange,CoreRangeSet> &core_spec, uint32_t initial_value);
+uint32_t CreateSemaphore(Program &program, const std::variant<CoreRange,CoreRangeSet> &core_spec, uint32_t initial_value, CoreType core_type=CoreType::WORKER);
```

Comment:
plz update doc table above for the new arg and also docs/source/tt_metal/apis/host_apis/buffers/CreateSemaphore.rst for html documentation